crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -539,13 +532,16 @@ impl<'src> Parser<'src> {
                 range: tok_range,
             })
         } else {
+            // TODO(dhruvmanila): This recovery isn't possible currently because the soft keyword
+            // transformer will always convert the `type` token to a `Name` token if it's not
+            // followed by a `Name` token.
             self.add_error(
                 ParseErrorType::OtherError(format!("expecting identifier, got {tok}")),
                 tok_range,
             );
-            #[allow(deprecated)]
-            Expr::Invalid(ast::ExprInvalid {
-                value: self.src_text(tok_range).into(),
+            Expr::Name(ast::ExprName {
+                id: String::new(),
+                ctx: ExprContext::Invalid,
                 range: tok_range,
             })
```

Comment:
Creating an empty `name` with context `Invalid` is valid here but we shouldn't consume the token in that case.